ci_scripts/prep-release

```diff
@@ -2,36 +2,54 @@
 
 set -e
 
-# Update the target database snapshot with the latest targets
+INITIAL_HEAD=$(git rev-parse HEAD)
+printf "HEAD before auto generating docs and target_database = ${INITIAL_HEAD}\n"
+
+printf "Attempting to update the target database snapshot with the latest changes\n"
 python ci_scripts/sync_board_database.py -vvv
 git add src/mbed_tools/targets/_internal/data/board_database_snapshot.json
 git add news/
 
-# Generate API docs and add license headers to them
+printf "Generating API docs and adding license headers\n"
 generate-docs --output_dir docs/api
 pre-commit run licenseheaders --all-files > /dev/null 2>&1 || true
 git add docs/api
 
 if ! NEW_VERSION=$(python ci_scripts/bump_version.py -vvv -n news/); then
-    printf "Nothing to do. Exiting.\n"
+    printf "No news files detected. Exiting.\n"
     exit 0
 fi
 
-# Commit and tag new version
-# Towncrier will fail if it detects a news file that has not been committed to
-# the repository. So we need to ensure the news file generated by
-# sync_board_database.py was committed before running towncrier.
-git commit -m "Prepare mbed-tools release ${NEW_VERSION}"
+# Update the index so we can check any recently staged changes
+git update-index -q --ignore-submodules --refresh
+
+if ! git diff-index --cached --quiet HEAD --ignore-submodules --
+then
+    printf "Found staged changes to commit after docs and target database update: $(git diff-index --cached --name-status -r --ignore-submodules HEAD)\n"
```

Comment:
What does `-r` do? I don't see it in the man page. Is it `--relative` or `--raw` or something else? I'm looking to understand the difference between what we print here and what is in our `if` above, so ensure we print what we are checking (other than purely cosmetic options).

Reply:
`-r` should be `--raw`, but `--raw` seems to be default now (and they don't document `-r` anymore). I'll remove this as it's not needed.